AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/HubEvents/AuthHubEventHandler.swift

```diff
@@ -0,0 +1,124 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSPluginsCore
+
+class AuthHubEventHandler: AuthHubEventBehavior {
+
+    var lastSendEventName: HubPayloadEventName?
+
+    init() {
+        setupHubEvents()
+    }
+
+    func sendUserSignedInEvent() {
+        dispatchAuthEvent(HubPayload.EventName.Auth.signedIn)
+    }
+
+    func sendUserSignedOutEvent() {
+        dispatchAuthEvent(HubPayload.EventName.Auth.signedOut)
+    }
+
+    func sendUserDeletedEvent() {
+        dispatchAuthEvent(HubPayload.EventName.Auth.userDeleted)
+    }
+
+    func sendSessionExpiredEvent() {
+        dispatchAuthEvent(HubPayload.EventName.Auth.sessionExpired)
+    }
+
+    // swiftlint:disable cyclomatic_complexity
+    private func setupHubEvents() {
+
+        _ = Amplify.Hub.listen(to: .auth) {[weak self] payload in
```

Comment:
I think listeners are for users. Amplify should only publish/dispatch the events. Maybe this is specific to iOS Hub implementation that I misunderstood. Please verify.

Reply:
So that is true. This specific case, there are some additional events that `AWSCognitoAuthPlugin` is dispatching. As an example when an event is dispatched from `signInAPI`, we listen to the event and validate if sign in was successful and then dispatch another event that has been exposed to the customer, which is this case is `HubPayload.EventName.Auth.signedIn`.. So to summarize, there are some events that are `internal` and some that are `public`. 

In this implementation we are mimicking the same behaviour as before, listening to the `internal` events and dispatching `public` events .. Hope this clarifies the behaviour.